File: 卡码/代码随想录/1/06-二叉树/Go/04-二叉搜索树属性/LeetCode_0098.go
//98. 验证二叉搜索树
package main

import (
	. "binaryTree/types"
)

func isValidBST(root *TreeNode) bool {
	var pre *TreeNode=nil
	var traversal func(node *TreeNode) bool
	traversal=func(node *TreeNode) bool{
		if node==nil {
			return true
		}
		left:=traversal(node.Left)
		if pre!=nil&&pre.Val>=node.Val {
			return false
		}
		pre=node
		right:=traversal(node.Right)
		return left&&right
	}
	return traversal(root)
}

func main(){

}

File: 卡码/代码随想录/Java/codeThoughts/src/main/java/com/kurumi/leetCode/codeThoughts/二叉树/_04_二叉搜索树属性/LeetCode_0098.java
package com.kurumi.leetCode.codeThoughts.二叉树._04_二叉搜索树属性;

import com.kurumi.leetCode.codeThoughts.entity.TreeNode;

/**
 * @Author kurumi
 * @Date 2026-07-03 上午11:52:20
 * @Description 98. 验证二叉搜索树
 */
public class LeetCode_0098 {
    static class Solution {
        TreeNode pre = null;

        public boolean isValidBST(TreeNode root) {
            if (root == null)
                return true;
            boolean left = isValidBST(root.left);
            if (pre != null && pre.val >= root.val)
                return false;
            pre = root;
            boolean right = isValidBST(root.right);
            return left && right;
        }
    }
}
